src/rgw/rgw_auth.cc

```diff
@@ -508,3 +508,44 @@ RGWAuthApplier::aplptr_t RGWKeystoneAuthEngine::authenticate() const
 
   return nullptr;
 }
+
+rgw::auth::Engine::result_t
+rgw::auth::Strategy::authenticate(const req_state* const s) const
+{
+  for (const stack_item_t& kv : auth_stack) {
+    const rgw::auth::Engine& engine = kv.first;
+    const auto& policy = kv.second;
+
+    auto&& res = engine.authenticate(s);
```

Comment:
i think `auto&&` makes perfect sense here. because we're returning this result by value, the caller will already require a move construction to store the result. changing this to `auto res =` would incur an extra call to the move constructor

p.s. the `return std::move(res);` below should be equivalent to `return res;` in either case